### PR TITLE
DB: Store recent validator stats

### LINKED ISSUE #98: Validators with low forging ratio should be dropped
Someone with enough voting power could introduce multiple stale validators and disrupt the network. Also an attack on several nodes, abandoned or shutdown validators could affect transaction processed by the network.

Validators with low performance should be droped from the top of validator list. After n blocks validators will be whitelisted again, if they fail again to perform, drop time doubles.

### PR DESCRIPTION
**Pull request by semuxdev**
This is primarily based on #665, to resolve #98 and #462.

### CODE REVIEW

**Comment by cryptokat** [comment]
Profiling result:
selection_346

**Comment by cryptokat** [comment]
Do we really need to store `RecentValidatorStats` in the database? Considering that most of overhead comes from encoding & decoding of `RecentValidatorStats` data structure, we can actually store the stats in a memory cache as `RECENT_VALIDATOR_STATS_DURATION` is kept low.

File: src/main/java/org/semux/core/BlockchainImpl.java [hunk 1]
```diff
@@ -493,10 +505,24 @@ public ValidatorStats getValidatorStats(byte[] address) {
         return (value == null) ? new ValidatorStats(0, 0, 0) : ValidatorStats.fromBytes(value);
     }
 
+    @Override
+    public RecentValidatorStats getRecentValidatorStats(byte[] address) {
+        ByteArray key = ByteArray.of(address);
+
+        RecentValidatorStats stats = recentValidatorStats.getIfPresent(key);
+        if (stats == null) {
+            stats = new RecentValidatorStats();
```
Comment (cryptokat) [inline src/main/java/org/semux/core/BlockchainImpl.java]: @semuxdev 
Sorry for the wrong information. My previously suggested approach doesn't really work well as there isn't an efficient way to recover stats at a certain height. 

I just come up with an alternative approach that could work. Please let me draft another pull request for that.

Comment (semuxdev) [reply src/main/java/org/semux/core/BlockchainImpl.java]: Agreed, it's also my concern.

I was thinking of adding the following index
```
block_number => primary validator
```,
so we can compute the recent stats easily.

**Comment by semuxdev** [comment]
Deprecated until we have better design.

**Pull request closed without merging**